### PR TITLE
Add a model connectivity tool

### PR DESCRIPTION
**Pull request by dangunter**
## Summary/Motivation:

Working with a stakeholder who requested this

## Changes proposed in this PR:
- Create a tool to do connectivity
- It also generates mermaid graphs
- Test it internally and against UKy flowsheet

## Reviewer's checklist / merge requirements:
- [ ] The head branch (i.e. the "source" of the changes) is not the `main` branch on the PR author's fork
- [ ] Documentation
- [ ] Tests
- [ ] Diagnostic tests for models

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: src/prommis/conn/connectivity.py [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+##############
+# Classes
+##############
+
+
+class OutputFormats(enum.Enum):
+    markdown = "markdown"
+    html = "html"
+    mermaid = "mermaid"
+    csv = "csv"
```
Comment (lbianchi-lbl) [inline src/prommis/conn/connectivity.py:65]: To resolve the Pylint errors for these class attributes, you can add the following to `pyproject.toml` after 

```
class-const-naming-style = "any"
```

More info: https://pylint.readthedocs.io/en/v2.17.7/user_guide/messages/convention/invalid-name.html#predefined-naming-styles

File: src/prommis/conn/connectivity.py [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+                f.write(
+                    "import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';\n"
+                )
+                f.write(
+                    "mermaid.initialize({securityLevel: 'loose', maxEdges: 2000});\n"
+                )
+                f.write("await mermaid.run();\n")
+                f.write("</script></body></html>")
+            case _:
+                raise ValueError(f"Bad output format: {output_format}")
```
Comment (lbianchi-lbl) [inline src/prommis/conn/connectivity.py:100]: As much as it's cool to see the pattern match syntax being used, this requires Python 3.10+ and thus won't be available to us as long as we have to support Python 3.9 (about 1 year more at least).

File: src/prommis/conn/connectivity.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+        return self._c
+
+    def _build_units(self):
+        units = {}
+        c1, c2 = 1, -1
+        for s in self._header[1:]:
+            abbr = "Unit_"
+            if c2 > -1:
+                abbr += chr(ord("A") + c2)
+            abbr += chr(ord("A") + c1)
```
Comment (lbianchi-lbl) [inline src/prommis/conn/connectivity.py:181]: I don't know what `chr()` or `ord()` do off the top of my head, so I'd suggest to add a comment to clarify.

File: src/prommis/conn/connectivity.py (hunk 2 above)
Comment (dangunter) [reply src/prommis/conn/connectivity.py:100]: yeah it was fun while it lasted

File: src/prommis/conn/connectivity.py (hunk 3 above)
Comment (dangunter) [reply src/prommis/conn/connectivity.py:181]: ah kids today. ok

File: src/prommis/conn/connectivity.py (hunk 1 above)
Comment (dangunter) [reply src/prommis/conn/connectivity.py:52]: I decided all-caps was OK for them anyways, but thanks for the pointer

**Comment by lbianchi-lbl** [comment]
@dangunter is working with @codytodonnell to add new functionality (integration with a drawing tool) to the code being added, so I've changed this back to draft.

**Comment by dangunter** [comment]
This has been moved to its own repo in the prommis org https://github.com/prommis/idaes-model-conn

**Pull request closed without merging**